src/frontend/screens/MapScreen/GPSPermissions/GPSPermissionsModal.tsx

```diff
@@ -1,34 +1,42 @@
 import React, {useEffect, useState} from 'react';
-import {GPSPermissionsDisabled} from './GPSPermissionsDisabled';
+import {GPSForegroundPermissionDisabled} from './GPSForegroundPermissionDisabled';
 import {GPSPermissionsEnabled} from './GPSPermissionsEnabled';
 import * as Location from 'expo-location';
 import {useGPSModalContext} from '../../../contexts/GPSModalContext';
 import {useTabNavigationStore} from '../../../hooks/useTabNavigationStore';
 import {BottomSheetModal, BottomSheetView} from '@gorhom/bottom-sheet';
 import {TAB_BAR_HEIGHT} from '../../../Navigation/Stack/AppScreens';
 import {StyleSheet} from 'react-native';
-import {useFocusEffect} from '@react-navigation/native';
+import {GPSBackgroundPermissionDisabled} from './GPSBackgroundPermissionDisabled';
 
 export const GPSPermissionsModal = React.memo(() => {
   const {setCurrentTab} = useTabNavigationStore();
   const [backgroundStatus] = Location.useBackgroundPermissions();
   const [foregroundStatus] = Location.useForegroundPermissions();
 
-  const [isGranted, setIsGranted] = useState<boolean | null>(null);
+  const [foregroundStatusGranted, setForegroundStatusGranted] = useState<
+    boolean | null
+  >(null);
+  const [backgroundStatusGranted, setBackgroundStatusGranted] = useState<
+    boolean | null
+  >(null);
```

Comment:
probably missing some context, but can these not be derived from the `backgroundStatus` and `foregroundStatus`?

Reply:
yeah, you would think. But its actually a [problem with the hooks](https://github.com/expo/expo/issues/30351). The hooks are not actually responsive, so hence why I am forcing the state here. I actually think that we should refactor the entire permissions landscape because this exposed some really annoying edge cases